### ISSUE TITLE
Support nested resources

### ISSUE THREAD

**Issue by carlosafonso** [carlosafonso/laravel-crud#4]
Nested resources should also be supported and managed by the library:

```
/foos/3/bars
```

**Comment by namelivia** [comment]
Having this one done would be pretty nice.

**Comment by namelivia** [comment]
Laravel already has support for [nested resources](https://laravel.com/docs/5.1/controllers#restful-nested-resources):
However, the controller methods are slightly different and its parameters also differ, so the CrudController would be no longer compatible.

I'm thinking on making a NestedCrudController keeping some of the main ideas of the CrudController like automatic picking of related models (this time would be `parentModel` and `childModel`) and most of the CRUD methods would be also reimplemented.

The downside of this is that at first there is going to be some code duplication, but that can probably be fixed with proper design patters application.

**Comment by namelivia** [comment]
There is also an issue when extracting the model names from the controller name. In the CrudController, if a controller is named `FoosController` the related model name would be `Foo` by doing some string operations.

In the case of a nested controller, the controller should be named `FooBarController`, so there would be the need for not only doing the same string operations that apply to the CrudController, that would set the model name as `FooBar` but also a split assuming a proper camelization of the name, for getting `Foo` and `Bar` as parent and child model names. If the camelization is now well done and a number != 2 of model names are extracted, maybe an Exception should be raised.

**Comment by carlosafonso** [comment]
Yeah this one's tricky.

It also depends on what URIs we are expecting to handle. Creating a nested resource route for `foos.bars` produces routes such as GET /foos/{fooId}/bars/{barId}. In this example, fooId is not really useful. The same applies to PUT and DELETE requests.

If we want to do this our way then a nested resource controller only differs from a regular one in the `index()` and `store()` methods, as they need to receive the ID of the parent entity.

But then we need the developer to carefully set up the correct routes. Maybe we can help with that too with some sort of `CrudRoute::resource()` method.

Although this might be overkill.

**Comment by namelivia** [comment]
You mean creating nested routes but instead of using the methods produced by the resource `foo.bars` allowing only index and store for them?

**Comment by carlosafonso** [comment]
Yes and no. Declaring a nested resource route will make the route work with the following action, among others:

``` php
/* route: GET /parent/{parentId}/entity/{entityId} */
public function show($parentId, $entityId)
{
    // ...
}
```

We could try to adapt the `CrudController` so that it can handle this action, but what's the point? We don't really need `$parentId`, do we? So we could just use the default `show()` for the entity.

**Comment by namelivia** [comment]
Yeah that is true, but for the case of the `index` you would need the `parentId`, in the case of the update or the delete you don't need it. But we were not reusing the `foo.bar` routes then already provided by Laravel.

**Comment by carlosafonso** [comment]
Yes, `index` does require the parent ID. I would do something like this:

``` php
Route::get('foos/{fooId}/bars'); // new
Route::get('bars/{barId}'); // standard
Route::post('foos/{fooId}/bars'); // new
Route::put('bars/{barId}'); // standard
Route::delete('bars/{barId}'); // standard
```

But this requires extra work on the developer. Maybe we could save all this work by using a helper which declares all these routes for you.

**Comment by namelivia** [comment]
Sure, if we are declaring those custom routes the helper is absolutely necesary.
